### PR TITLE
feat: [CI-2928]: Pre-populate default branch of the connector repo in Codebase input form

### PR DESCRIPTION
**Pull request by vardanbansal-harness**
### Summary

✅ This PR defaults to "branch" type build and sets the default branch of the repository configured in the codebase connector as the branch to build.

❗️This behaviour only applies to Codebase form in Run pipeline and **NOT** for triggers, input sets and templates.

https://user-images.githubusercontent.com/48940704/179683423-408eb9c9-1f37-4add-a694-ca3185c65c93.mov

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/utils/__tests__/CIUtils.test.ts [hunk 1]
```diff
@@ -76,4 +83,34 @@ describe('Test CIUtils', () => {
       )
     ).toBe(JSON.stringify([{ label: 'val1', value: 'val1' }]))
   })
+
+  test.only('getgetCodebaseRepoNameFromConnector method', () => {
+    const codebaseConnector: ConnectorInfoDTO = {
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/utils/__tests__/CIUtils.test.ts:88]: test.only?

Comment (vardanbansal-harness) [reply src/modules/70-pipeline/utils/__tests__/CIUtils.test.ts]: removed it, added it for testing

File: src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx [hunk 2]
```diff
@@ -273,8 +299,35 @@ function CICodebaseInputSetFormInternal({
...
+    (): string => get(formik?.values, codeBaseInputFieldFormName.repoName),
+    [get(formik?.values, codeBaseInputFieldFormName.repoName)]
+  )
+
+  const updateRepoName = useCallback(
+    (repoName: AcceptableValue | undefined) => formik.setFieldValue(codeBaseInputFieldFormName.repoName, repoName),
+    []
+  )
+
+  const debouncedFetchBranches = React.useRef(debounce(updateRepoName || noop, 2000)).current
```
Comment (vardanbansal-harness) [inline src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx:317]: On repo name input by a user for a git account level connector, we need to refetch the branches for the repo specified.

File: src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx [hunk 3]
```diff
@@ -323,14 +377,72 @@ function CICodebaseInputSetFormInternal({
...
     }
   }, [loadingConnectorDetails, connectorDetails])
 
+  useEffect(() => {
+    // Default branch needs to be set only if not specified by the user already for "branch" type build, only on Run Pipeline form
+    if (
+      !get(formik?.values, codeBaseInputFieldFormName.branch) &&
+      !isDefaultBranchSet &&
+      codeBaseType === CodebaseTypes.branch &&
+      codebaseConnector &&
```
Comment (mtran7) [inline src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx:395]: Can you test if codebaseConnector is and is not a runtime input?

Comment (vardanbansal-harness) [reply src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx]: Yes, tested for those cases.

File: src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx [hunk 4]
```diff
@@ -323,14 +377,72 @@ function CICodebaseInputSetFormInternal({
...
+              orgIdentifier,
+              projectIdentifier,
+              repoName: encodeURI(repoName),
+              size: 1
+            }
+          })
+            .then((result: ResponseGitBranchesResponseDTO) => {
+              setFetchingDefaultBranch(false)
+              formik.setFieldValue(codeBaseInputFieldFormName.branch, result.data?.defaultBranch?.name || '')
+              setIsDefaultBranchSet(!isEmpty(result.data?.defaultBranch?.name))
```
[the remark this answers is not in the document]
Comment (vardanbansal-harness) [reply src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx]: Done, cleaned it up.

**Review by mtran7 — APPROVED** [review]

**Review by avinashmadhwani02 — APPROVED** [review]

**Pull request merged**